bundles/tools.vitruv.applications.umljava/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaMethod.reactions

```diff
@@ -187,6 +187,7 @@ reaction UmlFeatureMadeStatic {
 routine setStatic(uml::Feature uFeat) {
     match {
         val jMod = retrieve java::AnnotableAndModifiable corresponding to uFeat
+			with !(uFeat instanceof org.eclipse.uml2.uml.Property && jMod instanceof ClassMethod)
```

Comment:
When matching a generic uml feature to a generic java object without specifying a tag, it may happen for the uml attribute case that the java attribute and the getter and setter are matched, as the tags are ignored. Therefore, this workaround is used to not match accessors. However, it would be cleaner if the language allows to specify tags to ignore while matching. @HeikoKlare I did not find any language construct that supports this. Is this possible? I.e. ```not tagged with ["a", "b"]```

Reply:
Currently, there is no language construct to do this, but it should be possible to add it if it's necessary.
If it is only about finding corresponding elements for which the correspondence is not tagged at all, you could try 
`retrieve X corresponding to Y tagged with ""`, as filtering for the tag `""` should only return corresponding elements, for which not tag was assigned on correspondence creation (if I understand the code correctly).

Otherwise, I would propose to add further correspondences with different tags between the same elements, as there is obviously some additional semantic relation between the elements.

Reply:
Thanks for the clarification. I already discovered the ```tagged with ""``` option but it seems pretty _hacky_ to me. Furthermore I did not chose this option as it restricts the matching more than intended. I want to exclude 2 specific tags and not restrict to no tag at all.
From my perspective it would make sense to add the ```not tagged with [x, y]``` functionality. However, I can't estimate the effort of doing so. Therefore, if this is considered as a reasonable addition, I would propose to leave this pull request as-is and add an issue describing the language addition and linking to this source code to be refactored when the language is ready.

Reply:
From my point of view, checking for not having a specific tag is not reasonable (in this case). It makes the assumption that every corresponding element that is not tagged with one of the given tags shares a certain value (in this case the value of `static`). If this is not, by definition, given for all corresponding elements of a given type, the corresponding elements of interest should be tagged (inclusively) instead of tagging the others (exclusively). But maybe someone has strong arguments what that's a bad idea.

Regarding this specific situation: I have to admit that I have not completely understood the problem. You wrote that accessors for attributes shall not be matched, but if it's about setting the `static` value of some attribute, it makes no sense to keep accessors as instance methods when the according attribute is static, or am I mistaken?

Reply:
Eventually, I split the method into separate methods for fields and methods. Using this, the class of the correspondence is further constrained such that I only have one match. However, I only applied this to the discussed method (as the other routines match so generic that it would require too much time for this PR to refactor them). For the remaining ones, I used the proposed ```tagged with ""``` approach.

Regarding the handling of accessors for static attributes: I did not touch the logic but left a ```TODO``` comment. This was done as the handling of accessors in such cases is not totally clear (remove them? / make them static? / re-add them again when attribute becomes non-static?) and should be done as part of a larger rework of the accessors handling logic, as there are other incorrectly handled cases (e.g. currently a setter is preserved when a field becomes final).

Reply:
Regarding the handling of accessors for static attributes: I think that is fine for now and if we feel that we want to properly support making fields static and the other way round, we can add that behavior after the fact. As we previously discussed, just because we can keep it consistent, it doesn't mean we need to, as that depends on the underlying purpose for which the Vitruv application is used for. Whether to remove the getters/setters or to make them static depends on what the application is then used for.